pkg/controller/clusters/status.go

```diff
@@ -0,0 +1,84 @@
+package clusters
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/controller"
+)
+
+// UpdateControlPlaneStatus checks the current state of the Cluster's control plane and updates the
+// Cluster status information.
+func UpdateControlPlaneStatus(ctx context.Context, client client.Client, cluster *anywherev1.Cluster) error {
+	kcp, err := controller.GetKubeadmControlPlane(ctx, client, cluster)
+	if err != nil {
+		return errors.Wrapf(err, "getting kubeadmcontrolplane")
+	}
+
+	updateControlPlaneInitializedCondition(cluster, kcp)
+
+	return nil
+}
+
+// updateControlPlaneInitializedCondition updates the ControlPlaneInitialized condition if it hasn't already been set.
+// This condition should be set only once.
+func updateControlPlaneInitializedCondition(cluster *anywherev1.Cluster, kcp *controlplanev1.KubeadmControlPlane) {
+	// Return early if the ControlPlaneInitializedCondition is already "True"
+	if conditions.IsTrue(cluster, anywherev1.ControlPlaneInitializedCondition) {
+		return
+	}
+
+	if kcp == nil {
+		conditions.Set(cluster, controlPlaneInitializationInProgressCondition())
+		return
+	}
+
+	// We make sure to check that the status is up to date before using it
+	if kcp.Status.ObservedGeneration != kcp.ObjectMeta.Generation {
+		conditions.MarkFalse(cluster, anywherev1.ControlPlaneInitializedCondition, anywherev1.OutdatedInformationReason, clusterv1.ConditionSeverityInfo, "")
+		return
+	}
+
+	// Surface any errors from the KubeadmControlPlane, relating to control plane availability.
+	kcpCondition := getKubeadmControlPlaneConditionIfError(kcp, controlplanev1.AvailableCondition)
+	if kcpCondition != nil {
+		conditions.MarkFalse(cluster, anywherev1.ControlPlaneInitializedCondition, kcpCondition.Reason, kcpCondition.Severity, kcpCondition.Message)
```

Comment:
I'm ok with this for now but we might want to think about it a bit more
reasons are supposed to be machine readable, which means the should be part of our api, so our users know what to look for
here we are mirroring capi blindly, which means our user won't know about these reasons

Reply:
Okay, we'll remove this section for now as the goal is currently to track progress of lifecycle operations. We can need to surface errors, make them more specific. We'll may iterate over this to surface errors in the future.